sqlmesh/core/macros.py

```diff
@@ -297,6 +300,15 @@ def columns_to_types(self, model_name: TableName | exp.Column) -> t.Dict[str, ex
 
         return columns_to_types
 
+    def get_snapshot(self, model_name: TableName | exp.Column) -> t.Optional[Snapshot]:
```

Comment:
don't you need to normalize the model name here

Reply:
Good point; fwiw I think this section will be rewritten soon by @eakmanrq when he rebases to get these changes.

For now I've addressed this in https://github.com/TobikoData/sqlmesh/pull/1761/commits/e67912b6d38559ecfea70eae642936e7c6b956d5